### PR TITLE
[WIP] Shunt sheet browser

### PR DESCRIPTION
**Pull request by rmpowell77**
This builds on #40, so requires that to be merged first.

This is sort of a WIP, looking for feedback.  This will create a preview sheet for each stunt sheet on the left hand side.  Except it isn't drawing dots, isn't scaling, and isn't setting up scroll bars if there are too many sheets...  Besides that, it's pretty interesting change.

### CODE REVIEW

**Comment by MichaelTamaki** [comment]
👍 Thank you for the head start, I’ll look at this later. Can you change the PR branch from master to the branch in #40? That way it’ll be easier to view the changes.

it might be difficult to complete the stuntsheet previews right now as we haven’t even finished drawing dots on the main grapher yet. I think for now, you could focus on just having a stuntsheet list without the preview, or work on a different issue?

**Comment by rmpowell77** [comment]
Oh, this also contains drawing the dots -- though they aren't in the right place yet.  Think of this PR as an early preview to get feedback on the formatting.

**Comment by MichaelTamaki** [comment]
I've been thinking about actually saving the SVGs as pictures, which we use as the preview. Not sure if this is actually possible, but it would save the browser from having to render every single stuntsheet all the time.

File: src/components/grapher/PerSheetPreview.vue [hunk 1]
```diff
@@ -0,0 +1,78 @@
+<template>
+  <div class="persheetpreview">
+    <svg class="persheetpreview-svg"><g>
```
Comment (MichaelTamaki) [inline src/components/grapher/PerSheetPreview.vue:3]: I think for now, we should park the sheet preview feature. I feel that `Grapher` is going to be changing a lot in the coming months, so if we had both `Grapher` and `PerSheetPreview`, we'd have to worry about and modify both whenever there's a change.

For now, how about we just have a simple stuntsheet selector and add/remove functionality?

File: src/components/menu-left/MenuLeft.vue [hunk 2]
```diff
@@ -1,14 +1,68 @@
 <template>
   <div class="column is-narrow menu-left">
     <h3>Menu left</h3>
+    <section>
```
Comment (MichaelTamaki) [inline src/components/menu-left/MenuLeft.vue:4]: `section` and `footer` are good for semantics in websites, but for web applications they sometimes add unnecessary default styling. We should use `div` instead.

File: src/models/StuntSheet.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ import { loadContinuity } from './continuity/load-continuity';
  * Defines the positions/directions in a formation and the continuities
  * used to reach the next position.
  *
- * @property title          - High level description
+ * @property title          - Title of this StuntSheet
```
Comment (MichaelTamaki) [inline src/models/StuntSheet.ts]: For property descriptions, it's more helpful if you use words that are not the property name. In this case, using `Title of this Stuntsheet` to describe `title` doesn't really add any additional insight into what `title` is used for... After all, the property name is `title` inside the `Stuntsheet` model.

File: src/components/menu-left/MenuLeft.vue [hunk 4]
```diff
@@ -1,14 +1,68 @@
...
+      <PerSheetPreview
+        v-for="sheet in numberOfSheets"
+        :key="sheet + '-sheet'"
+        sheet="sheet"
+      />
+    </section>
+    <footer class="modal-card-foot">
+      <b-button
```
Comment (MichaelTamaki) [inline src/components/menu-left/MenuLeft.vue:12]: We could consider using a number input from Buefy instead, which is basically the same thing: https://buefy.org/documentation/numberinput

**Review by MichaelTamaki — CHANGES_REQUESTED** [review]
Found a fix to the scroll bar issue. It's in regards to flexbox, which is kinda tricky and takes a bit to understand. See this guide if you're interested in learning about it: https://css-tricks.com/snippets/css/a-guide-to-flexbox/

In `MenuLeft.vue`, add `overflow-y: auto;`
```scss
.menu-left {
  width: 150px;
  overflow-y: auto;
}
```

In `App.vue`, add `min-height: 0;` to `.app--bottom`:
```scss
.app--bottom {
  flex: 1 1;
  // eslint-disable-next-line max-len
  // https://stackoverflow.com/questions/34144972/flexbox-avoid-child-with-longer-height-than-parent-column-direction
  min-height: 0;
}
```

Also see my comment regarding parking the stuntsheet preview functionality for now. I'd suggest either making simple stuntsheet controls or working on a different issue.

**Comment by MichaelTamaki** [comment]
@rmpowell77 FYI I'm going to make a simple PR for selecting/adding/removing stunt sheets so that I can start working on the continuity editor after that. We can keep this PR so we can track whether or not it's feasible to create an SVG for each stuntsheet.

**Comment by MichaelTamaki** [comment]
https://github.com/calband/calchart-redesign/pull/68

**Comment by MichaelTamaki** [comment]
Will finalize later. Will get to this at a later date when higher priorities are done. https://github.com/calband/calchart-redesign/issues/34

**Pull request closed without merging**